lib/cable_ready/operation_builder.rb

```diff
@@ -24,10 +24,26 @@ def initialize(identifier)
     def add_operation_method(name)
       return if respond_to?(name)
       singleton_class.public_send :define_method, name, ->(*args) {
-        selector, options = nil, args.first || {} # 1 or 0 params
-        selector, options = options, {} unless options.is_a?(Hash) # swap if only selector provided
-        selector, options = args[0, 2] if args.many? # 2 or more params
-        options.stringify_keys!
+        if args.one? && args.first.respond_to?(:to_operation_options) && [Array, Hash].include?(args.first.to_operation_options.class)
+          case args.first.to_operation_options.class.name
```

Comment:
Ruby also lets you use class checks against the instance in a case statement. For example,

```ruby
case args.first.to_operation_options
when Array then ...
when Hash then ...
else ...
end
```

Reply:
I tried! `rake test` was throwing
```
ArgumentError: :to_operation_options returned an Array. Must be an Array or Hash.
```
I could not figure out why this was happening. Changing it to comparing against the string made the problem go away. I would **love** if you cracked the code.